### PR TITLE
Financial Connections: added the first Financial Connections native UI test

### PR DESCRIPTION
**Pull request by kgaidis-stripe**
## Summary

Added UI test for Financial Connections Native.

Next step will be exploring how we can run these more often + trigger an action for oncall (though only will do that after testing it on myself/Slack). 

Then, we can think of adding more UI tests if-needed.

## Testing

See test running: 

https://user-images.githubusercontent.com/105514761/208951417-a87e33c6-440a-4d92-82b4-0a3337593cb7.mov

I then ran `bitrise run ui-tests`:

```
+------------------------------------------------------------------------------+
|                          bitrise summary: ui-tests                           |
+---+---------------------------------------------------------------+----------+
|   | title                                                         | time (s) |
+---+---------------------------------------------------------------+----------+
| ✓ | Start Xcode simulator                                         | 2.65 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Set CONFIGURATION_BUILD_DIR                                   | 0.48 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Set CONFIGURATION_TEMP_DIR                                    | 0.45 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Set Bundler to use local vendor directory                     | 0.63 sec |
+---+---------------------------------------------------------------+----------+
| - | Git Clone Repository (Skipped)                                | 0.59 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Run Tuist                                                     | 27.21 sec|
+---+---------------------------------------------------------------+----------+
| - | Bitrise.io Cache:Pull (Skipped)                               | 0.45 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Bundler                                                       | 0.86 sec |
+---+---------------------------------------------------------------+----------+
| - | Bitrise.io Cache:Push (Skipped)                               | 0.50 sec |
+---+---------------------------------------------------------------+----------+
| - | Restore SPM Cache (Beta) (Skipped)                            | 0.52 sec |
+---+---------------------------------------------------------------+----------+
| ✓ | Xcode Test for iOS                                            | 2.4 min  |
+---+---------------------------------------------------------------+----------+
| ✓ | Xcode Test for iOS                                            | 15.8 min |
+---+---------------------------------------------------------------+----------+
| ✓ | Xcode Test for iOS                                            | 55.29 sec|
+---+---------------------------------------------------------------+----------+
| - | Deploy to Bitrise.io - Apps, Logs, Artifacts (Skipped)        | 0.73 sec |
+---+---------------------------------------------------------------+----------+
| - | Send notification if failed (Skipped)                         | 0.51 sec |
+---+---------------------------------------------------------------+----------+
| Total runtime: 19.7 min                                                      |
+------------------------------------------------------------------------------+


Submitting anonymized usage information...
For more information visit:
https://github.com/bitrise-io/bitrise-plugins-analytics/blob/master/README.md
Checking for new CLI version...

Checking for plugin (step) new version...

Checking for plugin (init) new version...

Bitrise build successful
```

### CODE REVIEW

File: Example/FinancialConnections Example/Project.swift [hunk 1]
```diff
@@ -29,7 +29,7 @@ let project = Project(
             productName: "FinancialConnectionsExample",
             bundleId: "com.stripe.example.Connections-Example",
             infoPlist: "FinancialConnections Example/Info.plist",
-            sources: "FinancialConnections Example/*.swift",
+            sources: "FinancialConnections Example/**/*.swift",
```
Comment (kgaidis-stripe) [inline Example/FinancialConnections Example/Project.swift:32]: This fixes a bug. During Tuist migration, the `Playground` folder was not included which maybe makes the Playground not run properly.

File: Example/FinancialConnections Example/Project.swift [hunk 2]
```diff
@@ -45,12 +45,39 @@ let project = Project(
             settings: .stripeTargetSettings(
                 baseXcconfigFilePath: "BuildConfigurations/FinancialConnections-Example"
             )
+        ),
+        Target(
+            name: "FinancialConnectionsUITests",
```
Comment (kgaidis-stripe) [inline Example/FinancialConnections Example/Project.swift:50]: Copied a lot of this logic (and other) from `PaymentSheetUITest` target

File: bitrise.yml [hunk 3]
```diff
@@ -377,6 +377,12 @@ workflows:
         - test_repetition_mode: retry_on_failure
         - maximum_test_repetitions: "5"
         - scheme: PaymentSheet Example
+    - xcode-test@4:
```
Comment (kgaidis-stripe) [inline bitrise.yml:380]: this adds the UI test to be run on nightly builds; we may or may not want this but should be OK

I still will be exploring running this more often, and having a Financial Connections specific trigger, but that's a separate issue

**Review by vardges-stripe — APPROVED** [review]
Very nice!

**Pull request merged**